myokit/tests/test_formats_importers.py

```diff
@@ -82,6 +88,145 @@ def sbml(fname):
         # Model with listOfInitialValues and unit with multiplier
         sbml('Noble1962-initial-assignments-and-weird-unit.xml')
 
+    def test_parameters(self):
+        # Test Case: Hodkin Huxley
+        # expected
+        parameters = [
+            'V',
+            'V_neg',
+            'E',
+            'I',
+            'i_Na',
+            'i_K',
+            'i_L',
+            'm',
+            'h',
+            'n',
+            'E_R',
+            'Cm',
+            'g_Na',
+            'g_K',
+            'g_L',
+            'E_Na',
+            'E_K',
+            'E_L',
+            'V_Na',
+            'V_K',
+            'V_L',
+            'alpha_m',
+            'beta_m',
+            'alpha_h',
+            'beta_h',
+            'alpha_n',
+            'beta_n'
+        ]
+
+        # test whether parameters are in myokit model
+        for param in parameters:
+            assert self.HoHu.has_variable('sbml.' + param)
+
+    def test_units(self):
+        # Test Case: Hodkin Huxley
+        # expected
+        param_unit_dict = {
+            'V': myokit.units.V * 10 ** (-3),
+            'V_neg': myokit.units.V * 10 ** (-3),
+            'E': myokit.units.V * 10 ** (-3),
+            'I': None,
+            'i_Na': None,
+            'i_K': None,
+            'i_L': None,
+            'm': None,
+            'h': None,
+            'n': None,
+            'E_R': myokit.units.V * 10 ** (-3),
+            'Cm': None,
+            'g_Na': None,
+            'g_K': None,
+            'g_L': None,
+            'E_Na': myokit.units.V * 10 ** (-3),
+            'E_K': myokit.units.V * 10 ** (-3),
+            'E_L': myokit.units.V * 10 ** (-3),
+            'V_Na': myokit.units.V * 10 ** (-3),
+            'V_K': myokit.units.V * 10 ** (-3),
+            'V_L': myokit.units.V * 10 ** (-3),
+            'alpha_m': None,
+            'beta_m': None,
+            'alpha_h': None,
+            'beta_h': None,
+            'alpha_n': None,
+            'beta_n': None
+        }
+
+        # test whether parameters have correct units
+        for param in param_unit_dict:
+            unit = param_unit_dict[param]
+            assert unit == self.HoHu.get('sbml.' + param).unit()
+
+    def test_initial_values(self):
+        # Test Case: Hodkin Huxley
+        # expected
+        param_value_dict = {
+            'I': 0,
+            'E_R': -75,
+            'Cm': 1,
+            'g_Na': 120,
+            'g_K': 36,
+            'g_L': 0.3,
+            'E_Na': -190,
+            'E_K': -63,
+            'E_L': -85.613
+        }
+
+        # test whether parameters have correct initial values
+        for param in param_value_dict:
+            value = param_value_dict[param]
+            assert value == self.HoHu.get('sbml.' + param).value()
```

Comment:
One downside of doing it with lists and dicts is that if it fails you don't get nice output saying which entry fails, it'll just tell you line 184 failed, but that's not top priority for now :-)

Reply:
Yes, that's something I had to deal with a couple times now. How would you do it then? write out the asserts for each parameter explicitly? I'm happy to adapt for the next pul request.

Reply:
Exactly. It's not super important but the best way is to write explicit asserts I'm afraid. Maybe I should consider switching to pytest :D

Reply:
Ok I will leave it for now, and change in the next PR